src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c

```diff
@@ -964,8 +964,8 @@ static void ADIOI_LUSTRE_Fill_send_buffer(ADIO_File fd, const void *buf,
                                           int *done_to_proc, int iter, MPI_Aint buftype_extent)
 {
     /* this function is only called if buftype is not contig */
-    int i, p, flat_buf_idx, size;
-    int flat_buf_sz, buf_incr, size_in_buf, jj, n_buftypes;
+    int64_t i, p, flat_buf_idx, size;
+    int64_t flat_buf_sz, buf_incr, size_in_buf, jj, n_buftypes;
```

Comment:
Can we use `MPI_Aint`? 
And I would move the declaration of `i` and `p` to where it's being used. There is no way one can understand its use from the names.